pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2371,11 +2371,9 @@ private void updateTopicMessageDispatchRate() {
         this.pulsar().getExecutor().execute(() -> {
             // update message-rate for each topic
             forEachTopic(topic -> {
-                if (topic instanceof AbstractTopic) {
-                    ((AbstractTopic) topic).updateBrokerDispatchRate();
-                }
-                if (topic.getDispatchRateLimiter().isPresent()) {
-                    topic.getDispatchRateLimiter().get().updateDispatchRate();
+                if (topic instanceof PersistentTopic) {
```

Comment:
No `updateBrokerDispatchRate()` for non-persistent-topic ?

Reply:
Yes. Because `dispatchRateLimiter` is private in `PersistentTopic`, so I think it 's ok just updating `PersistentTopic` dispatch rate @Jason918

Reply:
But `updateBrokerDispatchRate()` is in `AbstractTopic` and works for non-persistent topic.
Further more, we can add dispatchRateLimiter for non-persistent-topic in the future.

Reply:
> Further more, we can add dispatchRateLimiter for non-persistent-topic in the future
I looks better if we add `updateDispatchRateLimiter()` as `AbstractTopic` method. @Jason918